### DISCUSSION TITLE
sdf support

### DISCUSSION THREAD

**Discussion by Jsevenh** [drfenixion/freecad.robotcad#44] [General]
Hello, have you ever thought about adding support for generating SDF files?

**Comment by drfenixion** [comment]
Hello, next feature i will work is adding sensors. It may require adding custom SDF map but only for sensor purposes.

What reason of generating SDF instead of URDF? Where is can be applicable?
Or do you mean to generate environment of robot as SDF?

**Comment by Jsevenh** [comment]
mostly for the environment  stuff ,

**Reply by drfenixion** [reply]
I dont have plans to make it near time. You can implement it by yourself.

**Reply by Jsevenh** [reply]
great, I'll have a look at how I can best implement it...,

**Reply by drfenixion** [reply]
Some thoughts on how to do it. You should export to SDF only not hided objects and ofc 
exclude robot links generated objects (Real, Visual, Collisions) from export.

**Reply by drfenixion** [reply]
Teoretically you can try to use current URDF generator and after that convert URDF to SDF.

**Reply by Jsevenh** [reply]
> Teoretically you can try to use current URDF generator and after that convert URDF to SDF.

so I'll  start with  having a look at the urdf generator code first, to get an I dea of the implementation and I how I can use it for sdf generation , instead of having to write another

**Reply by Jsevenh** [reply]
> You should export to SDF only not hided objects and ofc
> exclude robot links generated objects (Real, Visual, Collisions) from export.

I do not understand this

**Reply by drfenixion** [reply]
> I do not understand this

I mean you should exclude from export any generated objects related to Robot. Also you should exclude from export any hided (by user) objects.

**Reply by drfenixion** [reply]
Also it can be usefull implement export to SDF only selected objects.
It mean if something selected exporting it, if selected nothing exporting as i describe above.

**Comment by Jsevenh** [comment]
Lemme , make a plan then I'll get started over the weekend , in-case you have any suggestions you can share them , since you have plans on working  on sensor support

**Reply by drfenixion** [reply]
I will try to avoid SDF maps generating and use URDF special tags for Gazebo in sensors feature. But it is nice to have oportunity to change SDF file map and persist your generated SDF enviroment stuff. 

Idea how to do that it is make changeable SDF wrapper file and including SDF file with enviroment stuff (what you plan to generate).
SDF wrapper file let opportunity to add things or to switch to any needed options of SDF.
http://sdformat.org/tutorials?tut=composition#using-include-in-world

**Comment by Jsevenh** [comment]
I'll create a draft pull request once I get started ,
